client/windows/CMarketWindow.cpp

```diff
@@ -151,13 +154,13 @@ void CMarketWindow::createChangeModeButtons(EMarketMode currentMode, const IMark
 	if(isButtonVisible(EMarketMode::CREATURE_EXP))
 	{
 		addButton(AnimationPath::builtin("ALTSACC.DEF"), CGI->generaltexth->zelp[572], std::bind(&CMarketWindow::createAltarCreatures, this, market, hero), EShortcut::MARKET_CREATURE_EXPERIENCE);
-		if(marketWidget->hero->getAlignment() == EAlignment::GOOD)
+		if(marketWidget->hero->getAlignment() == EAlignment::GOOD && mode != EMarketMode::ARTIFACT_CREATURE_EXP)
```

Comment:
I just checked, the grotto in the city and the altar on the map work the same way. I believe that this change is necessary only for debugging?

Reply:
Yeah, you're right.

I assumed that all hero's alignment can do everything. But the grotto behaves like a normal altar of sacrifice on map. So I can greatly simplify this...